pages/instructorFileEditor/instructorFileEditorClient.js

```diff
@@ -7,6 +7,9 @@ window.InstructorFileEditor = function (options) {
     throw new Error(`Instructor file editor element ${options.elementId} was not found!`);
   }
 
+  const modelist = ace.require('ace/ext/modelist');
+  const aceMode = modelist.getModeForPath(options.fileName).mode;
```

Comment:
I saw you mention this in your other PR. I think it's pretty low-effort to do this on the server, so I'd rather keep that there to avoid the network + JS overhead associated with doing this on the client.